configure.ac

```diff
@@ -11,46 +12,67 @@ define(SECTION_TITLE,
 	echo '   $1   ' | sed -e's/./-/g' >&AS_MESSAGE_FD
 ])
 
+
 dnl works for checking define statements in c
 m4_include([m4/AX_CHECK_DEFINE.m4])
 
-freebsd_src=${FREEBSD_SRC:=/usr/src}
-
-dnl Setting appropriate CFLAGS
-
-CFLAGS="-nostdinc 
--I${freebsd_src}/cddl/lib/libumem 
--I${freebsd_src}/sys/cddl/compat/opensolaris/ 
--I${freebsd_src}/sys/cddl/compat/opensolaris 
--I${freebsd_src}/cddl/compat/opensolaris/include 
--I${freebsd_src}/cddl/compat/opensolaris/lib/libumem 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libzpool/common 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/common/zfs 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common/fs/zfs 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common/sys 
--I${freebsd_src}/cddl/contrib/opensolaris/head 
--I${freebsd_src}/sys/cddl/contrib/opensolaris/uts/common 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libnvpair 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libuutil/common 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libzfs/common 
--I${freebsd_src}/cddl/contrib/opensolaris/lib/libzfs_core/common 
-${CFLAGS} -I/usr/include"
-
-dnl Setting appropriate LDFLAGS
 LDFLAGS="-lzfs -lnvpair -lzfs_core -luutil -lgeom -v"
 
 
-AC_DEFINE(NEED_SOLARIS_BOOLEAN)
-AC_DEFINE(_XPG6)
+dnl We allow setting a zof variable which the users can set to force autoconf to consider the system is zof based
+
+AC_ARG_VAR(zof, AS_HELP_STRING([Force configure to run ZoL(f) checks]))
+AC_ARG_VAR(freebsd_src, AS_HELP_STRING([Provide path for freebsd source tree. This defaults to /usr/src/]))
+AC_ARG_VAR(zof_src, AS_HELP_STRING([Provide path for freebsd source tree. This defaults to /usr/local/]))
+
+zof_src=${zof_src:=/usr/local}
```

Comment:
But thats not zof_src, thats PREFIX,  to where headers are installed. See previous comment about --prefix in configure, thats common practice.